framework/auth/views.py

```diff
@@ -869,10 +889,20 @@ def external_login_email_post():
     destination = 'dashboard'
     for campaign in campaigns.CAMPAIGNS:
         if campaign != 'institution':
+            # Handle different url encoding schemes between `furl` and `urlparse/urllib`.
+            # OSF use `furl` to parse service url during service validation with CAS. However, `web_url_for()` uses
+            # `urlparse/urllib` to generate service url. `furl` handles `urlparser/urllib` generated urls while ` but
+            # not vice versa.
             campaign_url = furl.furl(campaigns.campaign_url_for(campaign)).url
             if campaigns.is_proxy_login(campaign):
-                campaign_url = furl.furl(web_url_for('auth_login', next=campaign_url, _absolute=True)).url
-            if service_url.startswith(campaign_url):
+                # proxy campaigns: OSF Preprints and branded ones
+                if check_service_url_with_proxy_campaign(service_url, campaign_url):
```

Comment:
Update ORCiD account linking/creation landing page for osf-preprints and branded preprints. Using regular expression instead of string comparison.